src/main/java/seedu/mindmymoney/command/CalculateInputCommand.java

```diff
@@ -0,0 +1,55 @@
+package seedu.mindmymoney.command;
+
+import seedu.mindmymoney.MindMyMoneyException;
+import seedu.mindmymoney.data.ExpenditureList;
+import seedu.mindmymoney.helper.GeneralFunctions;
+
+import static seedu.mindmymoney.constants.Indexes.INDEX_OF_FIRST_ITEM_IN_STRING;
+import static seedu.mindmymoney.constants.Flags.FLAG_OF_EXPENDITURE_PER_MONTH;
+import static seedu.mindmymoney.constants.Indexes.INDEX_OF_SECOND_ITEM_IN_STRING;
+import static seedu.mindmymoney.helper.Calculations.calculateExpenditurePerMonth;
+
+/**
+ * Represents the Calculate command.
+ */
+public class CalculateInputCommand extends Command {
+    private String calculateInput;
+    public ExpenditureList expenditureList;
+
+    public CalculateInputCommand(String calculateInput, ExpenditureList expenditureList) {
+        this.calculateInput = calculateInput;
+        this.expenditureList = expenditureList;
+    }
+
+    /**
+     * Indicates whether the program should exit.
+     *
+     * @return Indication on whether the program should exit.
+     */
+    @Override
+    public boolean isExit() {
+        return false;
+    }
+
+    /**
+     * Parses the input for calculate command.
+     *
+     * @throws MindMyMoneyException when inputs are invalid or flags are missing.
+     */
+    public void executeCommand() throws MindMyMoneyException {
+        try {
+            String[] parsedCalculateInput = GeneralFunctions.parseInput(calculateInput);
+            assert parsedCalculateInput[INDEX_OF_FIRST_ITEM_IN_STRING] != null
+                    : "First element in parsedCalculateInput is null";
+            switch (parsedCalculateInput[INDEX_OF_FIRST_ITEM_IN_STRING].toLowerCase()) {
```

Comment:
I think `if (parsedCalculateInput[INDEX_OF_FIRST_ITEM_IN_STRING].toLowerCase().equals(FLAG_OF_EXPENDITURE_PER_MONTH))` better communicates your intent here, since you're only checking one case in the `switch`. If line length is an issue, you can consider storing the string compared to `FLAG_OF_EXPENDITURE_PER_MONTH` in a separate variable first.

Reply:
this is for future calculate commands! so that we can append commands in later implementations